### PR TITLE
Issue/5846,4597,4027,4112

### PR DESCRIPTION
**Pull request by axel22**
Review by @jsuereth

### CODE REVIEW

File: src/library/scala/collection/Map.scala [hunk 1]
```diff
@@ -51,6 +51,7 @@ object Map extends MapFactory[Map] {
     def iterator                    = underlying.iterator
     override def default(key: A): B = d(key)
   }
+
```
Comment (jsuereth) [inline src/library/scala/collection/Map.scala:54]: needs a comment like: With this empty line, I fight injustice.

File: src/library/scala/collection/SortedMapLike.scala [hunk 2]
```diff
@@ -80,4 +90,9 @@ self =>
    */
   override def ++[B1 >: B](xs: GenTraversableOnce[(A, B1)]): SortedMap[A, B1] =
     ((repr: SortedMap[A, B1]) /: xs.seq) (_ + _)
+  
 }
+
+
```
Comment (jsuereth) [inline src/library/scala/collection/SortedMapLike.scala:96]: These empty lines can fight world hunger.

File: src/library/scala/collection/Map.scala (hunk 1 above)
Comment (axel22) [inline src/library/scala/collection/Map.scala:54]: where's the like button :)

Comment (lrytz) [reply src/library/scala/collection/Map.scala:54]: hey, they have it now on github!

**Pull request merged**